### DISCUSSION TITLE
Do casks support .dmg files ?

### DISCUSSION THREAD

**Discussion by Caesarovich** [Homebrew/discussions#3864] [Casks]
Hi, I would like to add a software which is downloadable as a .dmg file. Can casks do that ?

Sorry for the noobie question !

**Comment by p-linnane** [comment]
Yes they do. An example of a popular cask utilizing .dmg files is [`docker`](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/docker.rb).

**Comment by SMillerDev** [comment]
Yes, https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/extraterm.rb

**Comment by Caesarovich** [comment]
Thanks all for your answers. So here is where I am:
```rb
cask "lego-spike" do
  version "2.0.8"
  sha256 "a3738d0513d736918a6d71535ef3d85dd184af267c05698e49ac4c6b48f38e17"

  url "https://education.lego.com/_/downloads/SPIKE-PRIME_macOS_#{version}_Global.dmg"
  name "Lego SPIKE"
  desc "The intuitive SPIKE™ App lets your students develop coding capabilities with Scratch and Python, regardless of their skill level."
  homepage "https://education.lego.com/"

  app "Spike.app"
end

```
I am not sure about the 'app' part, how do I know what the name of the .app file inside of the .dmg ?

**Reply by gromgit** [reply]
> I am not sure about the 'app' part, how do I know what the name of the .app file inside of the .dmg ?

Download and open the DMG in Finder?

**Comment by Caesarovich** [comment]
Okay follow-up:
```rb
cask "lego-spike" do
  version "2.0.8"
  sha256 "46e2cbebc009f7ebe1766014f794aa29c5f13dffd903611a3f896bee8914e10e"

  url "https://education.lego.com/_/downloads/SPIKE-PRIME_macOS_#{version}_Global.dmg"
  name "Lego SPIKE"
  desc "The intuitive SPIKE™ App lets your students develop coding capabilities with Scratch and Python, regardless of their skill level."
  homepage "https://education.lego.com/"

  app "Spike.app"
end
```
For now this works, so now I would love if you guys could provide me details on some more fields I could add.

For example when creating projects in this app it saves them in  `/Library/Containers/com.lego.education.spike/Data/LEGO Education SPIKE/*`, Should I add a 'zap' rule for it to remove these files ?

**Reply by p-linnane** [reply]
I encourage you to read [Cask Cookbook](https://docs.brew.sh/Cask-Cookbook) which explains all the parts of a cask. It also includes options for creating `zap` stanzas.
